question20/main.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+const wrapper = document.getElementById('js-wrapper');
+
+const createElementWithClassName = (element, name) => {
+  const createdElement = document.createElement(element);
+  createdElement.classList.add(name);
+  return createdElement;
+};
+
+const showLoadImg = () => {
+  const loadImg = createElementWithClassName("img", "loading");
+  loadImg.src = "./img/loading-circle.gif";
+  loadImg.id = 'loading';
+  wrapper.appendChild(loadImg);
+} 
+
+const removeLoadImg = () => {
+  document.getElementById('loading').remove();
+} 
+
+const getData = new Promise((resolve) => {
+  showLoadImg();
+  setTimeout(() => {
+    resolve(fetch('https://mocki.io/v1/1f12787c-e5e9-4cfa-9c1d-2331884ca8ec'));
+  }, 3000);
+})
+
+//Get json data
+const callApi = async() => {
+  try {
+    const response = await getData;
+    if (!response.ok) {
+      console.error(`${response.status}:${response.statusText}`);
+    }
+    const json = await response.json();
+    return json.data;
+  }
+  catch(error) {
+    console.error(error);
+  }
+  finally {
+    removeLoadImg();
+  }
+}
+
+const init = async() => {
+  const userData = await callApi();
+  if (userData){
+    renderTabaleElement(userData);
+    } else {
+    wrapper.textContent = "データの表示に失敗しました";
+    console.error('データの表示に失敗しました'); 
+  }
+}
+init();
+
+const userDataItems = {
+  userId: "ID",
+  name: "名前",
+  gender: "性別",
+  age: "年齢"
+}
+
+const renderTabaleElement = (userData)  => {
+  const table = createElementWithClassName("table", "table");
+  const tbody = createElementWithClassName("tbody", "tbody");
+  const userDataKeys = Object.keys(userDataItems);
+  const tableHeadValue = Object.values(userDataItems);
+  wrapper.appendChild(table).appendChild(tbody).appendChild(getTableHeadElement(tableHeadValue));
+  wrapper.appendChild(table).appendChild(tbody).appendChild(getTableDataElement(userData,userDataKeys));
+}
+
+const getTableHeadElement = (tableHeadValue) => {
+  const tr = createElementWithClassName("tr", "tr");
+  const fragmentThElement = document.createDocumentFragment();
+  for (const value of tableHeadValue) {
+    const th = createElementWithClassName("th", "th");
+    th.textContent = value;
+    fragmentThElement.appendChild(tr).appendChild(th);
+  }
+  return fragmentThElement;
+}
```

Comment:
IMO
I see that the return value is fragmentThElement even though the function is to get the TableHeadElement.

Why don't you change the function name or change the content to return thead?

[thead](https://developer.mozilla.org/ja/docs/Web/HTML/Element/thead)

Reply:
Thank you for your review and reference! Your right..! I changed the content to return thead. What do you think?
fixed commit: https://github.com/csakyo/homework/pull/34/commits/3028c43f1484797de03b65f0147907c561624f1b